include/boost/process/v2/detail/impl/exe.ipp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022 Klemens D. Morgenstern
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+#ifndef BOOST_PROCESS_V2_IMPL_DETAIL_EXE_IPP
+#define BOOST_PROCESS_V2_IMPL_DETAIL_EXE_IPP
+
+#include <boost/process/v2/detail/exe.hpp>
+
+BOOST_PROCESS_V2_BEGIN_NAMESPACE
+
+namespace detail
+{
+
+namespace ext
+{
+
+#if defined(_WIN32)
+
+BOOST_PROCESS_V2_DECL
+HANDLE open_process_with_debug_privilege(pid_type pid) {
+    HANDLE proc = nullptr;
+    HANDLE hToken = nullptr;
+    LUID luid;
+    TOKEN_PRIVILEGES tkp;
+    if (OpenProcessToken(GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, &hToken)) {
+        if (LookupPrivilegeValue(nullptr, SE_DEBUG_NAME, &luid)) {
+            tkp.PrivilegeCount = 1;
+            tkp.Privileges[0].Luid = luid;
+            tkp.Privileges[0].Attributes = SE_PRIVILEGE_ENABLED;
+            if (AdjustTokenPrivileges(hToken, false, &tkp, sizeof(tkp), nullptr, nullptr)) {
+                proc = OpenProcess(PROCESS_ALL_ACCESS, false, pid);
+            }
+        }
+        CloseHandle(hToken);
+    }
+    return proc;
+}
+
+BOOST_PROCESS_V2_DECL
+BOOL is_x86_process(HANDLE proc) {
+    BOOL isWow = true;
+    SYSTEM_INFO systemInfo;
+    GetNativeSystemInfo(&systemInfo);
```

Comment:
That is correct, unless I missed something: https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getnativesysteminfo

Reply:
The function below it can fail, though, so i added a check.  I'm not sure what to do when that function fails, because if it returns a true, but the target process is actually x64 then it will likely segfault when reading proc info depending on the function